wallet/src/main/java/com/android/identity_credential/wallet/TransferHelper.kt

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.identity_credential.wallet
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import androidx.core.content.edit
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.android.identity.android.mdoc.deviceretrieval.DeviceRetrievalHelper
+import com.android.identity.android.mdoc.transport.DataTransport
+import com.android.identity.android.securearea.AndroidKeystoreSecureArea
+import com.android.identity.android.storage.AndroidStorageEngine
+import com.android.identity.credential.CredentialStore
+import com.android.identity.mdoc.connectionmethod.ConnectionMethod
+import com.android.identity.mdoc.response.DeviceResponseGenerator
+import com.android.identity.securearea.SecureAreaRepository
+import com.android.identity.storage.StorageEngine
+import com.android.identity.util.Constants
+import com.android.identity.util.Logger
+import java.io.File
+import java.security.KeyPair
+import java.security.PublicKey
+import java.util.OptionalLong
+
+class TransferHelper private constructor(private val context: Context) {
```

Comment:
Move this into `PresentationActivity` and create the `DeviceRetrievalHelper` from its `onCreate()` method... you want `NfcEngagementHelper` to pass the data

```
eDeviceKeyPair: KeyPair,
eDeviceKeyCurve: Int,
transport: DataTransport,
deviceEngagement: ByteArray,
handover: ByteArray
```

You can use `putExtra()` to do this in the `Intent` passed to `startActivity()`, and then get those extras in `onCreate()`.

This way all use of `DeviceRetrievalHelper` is kept to a single file which makes it much easier to reason about.